### PR TITLE
tokyo-night-gtk: init at 2023.01.17

### PR DESCRIPTION
**Pull request by garaiza-93**
###### Description of changes
https://github.com/Fausto-Korpsvart/Tokyo-Night-GTK-Theme

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by JulienMalka — CHANGES_REQUESTED** [review]
Hi, thanks for contributing. Could you format your change into 2 commits:
1) ``maintainers: add garaiza-93``
2) ``tokyo-night-gtk: init at 2023.01.17``
Thanks!

**Comment by garaiza-93** [comment]
sure thing, done!

File: pkgs/data/themes/tokyo-night-gtk/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    rev = "f7ae3421ac0d415ca57fb6224e093e12b8a980bb";
+    sha256 = "sha256-90V55pRfgiaP1huhD+3456ziJ2EU24iNQHt5Ro+g+M0=";
+  };
+
+  propagatedUserEnvPkgs = [
+    gtk-engine-murrine
+  ];
+
+  buildInputs = [
+    gnome-themes-extra
```
Comment (NickCao) [inline pkgs/data/themes/tokyo-night-gtk/default.nix:24]: Why add this to buildInputs?

Comment (SoraTenshi) [reply pkgs/data/themes/tokyo-night-gtk/default.nix:24]: > Why add this to buildInputs?

It's part of the installing theme description on the [GH Page](https://github.com/Fausto-Korpsvart/Tokyo-Night-GTK-Theme#installing-themes)

Comment (NickCao) [reply pkgs/data/themes/tokyo-night-gtk/default.nix:24]: But as you have set `dontBuild = true;`, adding it should have no visible effect.

Comment (SoraTenshi) [reply pkgs/data/themes/tokyo-night-gtk/default.nix:24]: > But as you have set `dontBuild = true;`, adding it should have no visible effect.

Makes sense, i guess `dontBuild = true` should be removed then.

Comment (NickCao) [reply pkgs/data/themes/tokyo-night-gtk/default.nix:24]: As this package is not actually built (just copying the files to output), I think removing buildInputs altogether is the right solution. Please try and test.

Comment (SoraTenshi) [reply pkgs/data/themes/tokyo-night-gtk/default.nix:24]: > As this package is not actually built (just copying the files to output), I think removing buildInputs altogether is the right solution. Please try and test.

my reasoning behind it is, that the already [existing catpuccin-gtk-theme](https://github.com/NixOS/nixpkgs/blob/nixos-22.11/pkgs/data/themes/catppuccin-gtk/default.nix) should also be adjusted, unless it's different?

Comment (NickCao) [reply pkgs/data/themes/tokyo-night-gtk/default.nix:24]: Trying building it with the patch below applied:
```
--- a/pkgs/data/themes/catppuccin-gtk/default.nix
+++ b/pkgs/data/themes/catppuccin-gtk/default.nix
@@ -40,7 +40,6 @@ stdenvNoCC.mkDerivation rec {
   nativeBuildInputs = [ gtk3 sassc ];

   buildInputs = [
-    gnome-themes-extra
     (python3.withPackages (ps: [ ps.catppuccin ]))
   ];
```
And there's no difference in the output according to `diff -r`
Yes, that package should also be adjusted, maybe together with many other theme packages.

Comment (garaiza-93) [reply pkgs/data/themes/tokyo-night-gtk/default.nix:24]: alright, i'll remove `dontBuild` and `buildInputs`.

File: pkgs/data/themes/tokyo-night-gtk/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{ lib
+, stdenvNoCC
+, fetchFromGitHub
+, gnome-themes-extra
```
Comment (NickCao) [inline pkgs/data/themes/tokyo-night-gtk/default.nix:4]: You should also remove the now unused param. And please squash the fixup commits.

Comment (garaiza-93) [reply pkgs/data/themes/tokyo-night-gtk/default.nix]: done

**Review by NickCao — APPROVED** [review]

File: pkgs/data/themes/tokyo-night-gtk/default.nix (hunk 1 above)
Comment (romildo) [reply pkgs/data/themes/tokyo-night-gtk/default.nix]: By removing the dependence from buildInputs, the dependence will have to be explicitly listed in the system configuration file. It is a runtime configuration.

Comment (NickCao) [reply pkgs/data/themes/tokyo-night-gtk/default.nix]: If it is a runtime dependency, it should be in `propagatedUserEnvPkgs` just like `gtk-engine-murrine` do, adding it to `buildInputs` shall has no observable effect (correct me if I am wrong).

Comment (romildo) [reply pkgs/data/themes/tokyo-night-gtk/default.nix]: @jtojnar do you know what is the right way of dealing with runtime theming dependences in nixpkgs?

Comment (jtojnar) [reply pkgs/data/themes/tokyo-night-gtk/default.nix]: @romildo There is not really a way outside of `propagatedUserEnvPkgs`, which is an ugly hack that will not work in some cases. Your best bet is adding a `passthru.parentThemes` or something and creating a NixOS module that deals with it. Some discussion in https://github.com/NixOS/nixpkgs/issues/73967, https://github.com/NixOS/nixpkgs/issues/39057….

**Pull request merged**